File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/contract/WhatIfMainContract.java
package xyz.jienan.xkcd.whatif.contract;

import java.util.List;

import xyz.jienan.xkcd.base.BaseView;
import xyz.jienan.xkcd.home.base.ContentMainBasePresenter;
import xyz.jienan.xkcd.model.WhatIfArticle;

public interface WhatIfMainContract {

    interface View extends BaseView<Presenter> {

        void latestWhatIfLoaded(WhatIfArticle whatIfArticle);

        void showFab(WhatIfArticle whatIfArticle);

        void toggleFab(boolean isFavorite);

        void showThumbUpCount(Long thumbCount);

        void renderWhatIfSearch(List<WhatIfArticle> articles);
    }

    interface Presenter extends ContentMainBasePresenter {
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/contract/WhatIfMainContract.kt
package xyz.jienan.xkcd.whatif.contract

import xyz.jienan.xkcd.base.BaseView
import xyz.jienan.xkcd.home.base.ContentMainBasePresenter
import xyz.jienan.xkcd.model.WhatIfArticle

interface WhatIfMainContract {

    interface View : BaseView<Presenter> {

        fun latestWhatIfLoaded(whatIfArticle: WhatIfArticle)

        fun showFab(whatIfArticle: WhatIfArticle)

        fun toggleFab(isFavorite: Boolean)

        fun showThumbUpCount(thumbCount: Long?)

        fun renderWhatIfSearch(articles: List<WhatIfArticle>)
    }

    interface Presenter : ContentMainBasePresenter {
        fun getBookmark() : Long

        fun setBookmark(index: Long) : Boolean
    }
}
